src/main/java/com/jcabi/github/Repos.java

```diff
@@ -98,4 +106,243 @@ void remove(@NotNull(message = "coordinates can't be NULL")
+        public String homepage() {
+            return this.home;
+        }
+
+        /**
+         * Auto-init the new repo?
+         * If absent, the GitHub default will be used.
+         * @return Optional boolean
+         */
+        @NotNull(message = "optional itself is never NULL")
+        public Optional<Boolean> autoInit() {
+            return this.init;
+        }
+
+        /**
+         * Returns a NewRepoSettings with the given name.
+         * The name cannot be empty.
+         * @param nom Name of the new repo
+         * @return NewRepoSettings
+         */
+        @NotNull(message = "renamed settings is never NULL")
+        public NewRepoSettings withName(
+            @NotNull(message = "name can't be NULL") final String nom
+        ) {
+            return new NewRepoSettings(
+                nom,
+                this.priv,
+                this.descr,
+                this.home,
+                this.init
+            );
+        }
+
+        /**
+         * Returns a NewRepoSettings with the given privacy.
+         * @param privacy Privateness of the new repo
+         * @return NewRepoSettings
+         */
+        @NotNull(message = "new repo settings is never NULL")
+        public NewRepoSettings withPrivacy(final boolean privacy) {
+            return new NewRepoSettings(
+                this.nam,
+                privacy,
+                this.descr,
+                this.home,
+                this.init
+            );
+        }
+
+        /**
+         * Returns a NewRepoSettings with the given description.
+         * @param desc Description
+         * @return NewRepoSettings
+         */
+        @NotNull(message = "adjusted settings is never NULL")
+        public NewRepoSettings withDescription(
+            @NotNull(message = "description can't be NULL") final String desc
+        ) {
+            return new NewRepoSettings(
+                this.nam,
+                this.priv,
+                desc,
+                this.home,
+                this.init
+            );
+        }
+
+        /**
+         * Returns a NewRepoSettings with the given homepage.
+         * @param homep Homepage URL
+         * @return NewRepoSettings
+         */
+        @NotNull(message = "changed settings is never NULL")
+        public NewRepoSettings withHomepage(
+            @NotNull(message = "homepage can't be NULL") final String homep
+        ) {
+            return new NewRepoSettings(
+                this.nam,
+                this.priv,
+                this.descr,
+                homep,
+                this.init
+            );
+        }
+
+        /**
+         * Returns a NewRepoSettings with the given auto-init enabledness.
+         * @param auto Auto-init the new repo?
+         * @return NewRepoSettings
+         */
+        @NotNull(message = "modified settings is never NULL")
+        public NewRepoSettings withAutoInit(final Optional<Boolean> auto) {
```

Comment:
@pinaf Well, I figure it's either this builder-ish pattern, or have the public constructor take many parameters even though most users will only care about a couple of them. I'm open to other ideas though.